rotkehlchen/chain/ethereum/airdrops.py

```diff
@@ -196,21 +214,43 @@ def get_airdrop_data(airdrop_csv_path: str, name: str, data_dir: Path) -> Iterat
             raise RemoteError(
                 f'File {filename} contains invalid data. Check logs.',
             ) from e
+
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.AIRDROPS_HASH, name),
+                value=airdrop_data.csv_hash,
+            )
     # Verify the CSV file
     with open(filename, encoding='utf8') as csvfile:
         iterator = csv.reader(csvfile)
         next(iterator)  # skip header
         yield from iterator
 
 
-def get_poap_airdrop_data(poap_airdrop_path: str, name: str, data_dir: Path) -> dict[str, Any]:
+def get_poap_airdrop_data(airdrop_data: list[str], name: str, data_dir: Path) -> dict[str, Any]:
+    """Returns a dictionary of POAP airdrop data after downloading it locally for the first time.
+    If a new JSON is found in the index, it will be downloaded again to updated the local JSON file
+    and yield new data."""
     airdrops_dir = data_dir / APPDIR_NAME / AIRDROPSPOAPDIR_NAME
     airdrops_dir.mkdir(parents=True, exist_ok=True)
     filename = airdrops_dir / f'{name}.json'
+
+    with GlobalDBHandler().conn.read_ctx() as cursor:
```

Comment:
There is a lot of common functionality between this function and the `get_airdrop_data` one. Think you can abstract the common stuff out?